### PR TITLE
Migrate properties panel example to 1.0.0-alpha

### PR DESCRIPTION
**Pull request by barmac**
This migrates the `properties-panel-extension` example to the v1.0.0-alpha.0 of the properties panel. Read through the code changes to learn how to perform the migration.

### CODE REVIEW

**Comment by barmac** [comment]
WIP until `bpmn-js-properties-panel@1.0.0-alpha.0` is released and integrated.

File: properties-panel-extension/README.md [hunk 1]
```diff
@@ -73,60 +67,77 @@ function MagicPropertiesProvider(propertiesPanel, translate) {
...
+  const translate = useService('translate');
+  const debounce = useService('debounceInput');
+
+  const getValue = () => {
+    return element.businessObject.spell || '';
   }
-};
-```
 
-You can look into the [`EntryFactory`](https://github.com/bpmn-io/bpmn-js-properties-panel/blob/master/lib/factory/EntryFactory.js) to find many other useful reusable form input components. You can also go further and define what happens if you enter text in an input field and what is shown in it if the element is selected. To do so you can override `entry#set` and `entry#get` methods. A good example for this is [`DocumentationProps`](https://github.com/bpmn-io/bpmn-js-properties-panel/blob/master/lib/provider/bpmn/parts/DocumentationProps.js).
```
Comment (MaxTru) [inline properties-panel-extension/README.md:126]: Great we can drop all this :-)

File: properties-panel-extension/app/provider/magic/MagicPropertiesProvider.js [hunk 2]
```diff
@@ -1,31 +1,16 @@
-// require your custom property entries.
+// Require your custom property entries.
```
Comment (MaxTru) [inline properties-panel-extension/app/provider/magic/MagicPropertiesProvider.js:1]:
```suggestion
// Import your custom property entries
```

**Comment by MaxTru** [comment]
Something is broken with the implementation (or we have an undiscovered bug?), the data marker does not show up:
Screenshot_20211220_082518

File: properties-panel-extension/README.md [hunk 3]
```diff
@@ -73,60 +67,77 @@ function MagicPropertiesProvider(propertiesPanel, translate) {
...
+// Create the custom magic group
+function createMagicGroup(element, translate) {
 
-  // Create a group called "Black Magic".
-  var blackMagicGroup = {
-    id: 'black-magic',
-    label: 'Black Magic',
-    entries: []
+  // create a group called "Magic properties".
+  var magicGroup = {
```
Comment (MaxTru) [inline properties-panel-extension/README.md:82]: We are using `const` already in various places in this example, hence I propose to also use it here

File: properties-panel-extension/app/provider/magic/MagicPropertiesProvider.js [hunk 4]
```diff
@@ -35,45 +20,53 @@ export default function MagicPropertiesProvider(propertiesPanel, translate) {
...
 
-MagicPropertiesProvider.$inject = [ 'propertiesPanel', 'translate' ]
\ No newline at end of file
+MagicPropertiesProvider.$inject = [ 'propertiesPanel', 'translate' ];
+
+// Create the custom magic group
+function createMagicGroup(element, translate) {
+
+  // create a group called "Magic properties".
+  var magicGroup = {
```
Comment (MaxTru) [inline properties-panel-extension/app/provider/magic/MagicPropertiesProvider.js:65]: We are using `const` in various places in this example, hence I propose to use it here as well

**Review by MaxTru — CHANGES_REQUESTED** [review]
Looks good. I found a few things to consider

**Comment by MaxTru** [comment]
> Something is broken with the implementation (or we have an undiscovered bug?), the data marker does not show up: Screenshot_20211220_082518

Okay I know, the `TextFieldEntry` misses its ID. 

So it needs to be like this to fix this I suppose:

```
  return <TextFieldEntry
    id="spell"
    element={ element }
    description={ translate('Apply a black magic spell') }
    label={ translate('Spell') }
    getValue={ getValue }
    setValue={ setValue }
    debounce={ debounce }
  />
```

File: properties-panel-extension/app/provider/magic/MagicPropertiesProvider.js (hunk 2 above)
Comment (barmac) [reply properties-panel-extension/app/provider/magic/MagicPropertiesProvider.js]: Fixed.

**Comment by barmac** [comment]
> > Something is broken with the implementation (or we have an undiscovered bug?), the data marker does not show up: Screenshot_20211220_082518
> 
> Okay I know, the `TextFieldEntry` misses its ID.
> 
> So it needs to be like this to fix this I suppose:
> 
> ```
>   return <TextFieldEntry
>     id="spell"
>     element={ element }
>     description={ translate('Apply a black magic spell') }
>     label={ translate('Spell') }
>     getValue={ getValue }
>     setValue={ setValue }
>     debounce={ debounce }
>   />
> ```

Fixed:

image

File: properties-panel-extension/README.md [hunk 5]
```diff
@@ -73,60 +67,77 @@ function MagicPropertiesProvider(propertiesPanel, translate) {
 
 ### Define a Group
 
-As part of the properties provider we define the groups for the magic tab, too:
+As part of the properties provider we define the magic group:
 
 ```javascript
 // Require your custom property entries.
```
Comment (MaxTru) [inline properties-panel-extension/README.md:73]: Analog to the other change in the sourcefile, here we should update as well

**Review by MaxTru — APPROVED** [review]

File: properties-panel-extension/README.md [hunk 6]
```diff
@@ -2,17 +2,16 @@
 
 This example shows how to extend the [bpmn-js-properties-panel](https://github.com/bpmn-io/bpmn-js-properties-panel) with custom properties.
```
Comment (nikku) [inline properties-panel-extension/README.md:3]: We could add a note on top of the README that it works differently in `properties-panel@0.x` (and same on [properties-panel](https://github.com/bpmn-io/bpmn-js-properties-panel) README).

Comment (barmac) [reply properties-panel-extension/README.md:3]: https://github.com/bpmn-io/bpmn-js-examples/pull/144

**Pull request merged**